File: src/android/org/sunbird/sync/model/Request.java
package org.sunbird.sync.model;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by swayangjit on 25/3/20.
 */
public class Request {
    private String host;
    private String path;
    private String type;
    private JSONObject headers;
    private Object body;
    private String serializer;
    private int noOfFailureSync;

    public Request(String host, String path, String type, JSONObject headers, String serializer, Object body) {
        this.host = host;
        this.path = path;
        this.type = type;
        this.headers = headers;
        this.serializer = serializer;
        this.body = body;
    }

    public String getHost() {
        return host;
    }

    public String getPath() {
        return path;
    }

    public String getType() {
        return type;
    }

    public JSONObject getHeaders() {
        return headers;
    }

    public void setHeaders(JSONObject headers){
        this.headers = headers;
    }

    public Object getBody() {
        return body;
    }

    public String getSerializer() {
        return serializer;
    }

    public int getNoOfFailureSync() {
        return noOfFailureSync;
    }

    public void setNoOfFailureSync(int noOfFailureSync) {
        this.noOfFailureSync = noOfFailureSync;
    }

    @Override
    public String toString() {
        return "Request{" +
                "host='" + host + '\'' +
                ", path='" + path + '\'' +
                ", type='" + type + '\'' +
                ", headers=" + headers.toString() +
                ", body='" + body + '\'' +
                ", serializer='" + serializer + '\'' +
                '}';
    }

    public JSONObject toJSON() throws JSONException {
        JSONObject request = new JSONObject();
        request.put("host", host);
        request.put("type", type);
        request.put("path", path);
        request.put("headers", headers);
        request.put("serializer", serializer);
        request.put("body", body);
        request.put("noOfFailureSync", noOfFailureSync);
        return request;
    }
}

File: src/ios/model/request.swift
import Foundation

public class Request {
    var host: String
    var path: String
    var type: String
    var headers: [String: Any]
    var body: [String: Any]
    var serializer: String
    var noOfFailureSync: Int

    init(_ host: String, _ path: String, _ type: String, _ headers: [String: Any], _ serializer: String, _ body: [String: Any]){
        self.host = host
        self.path = path
        self.type = type
        self.headers = headers
        self.serializer = serializer
        self.body = body
    }

    func getHost() -> String {
        return self.host;
    }

    func getPath() -> String {
        return self.path;
    }

    func getType() -> String {
        return self.type;
    }

    func getHeaders() -> [String: Any] {
        return self.headers;
    }

    func setHeaders(_ headers: [String: Any]) {
        self.headers = headers;
    }

    func getBody() -> [String: Any] {
    return self.body;
    }

    func getSerializer() -> String {
        return self.serializer;
    }

    func getNoOfFailureSync() -> Int {
        return self.noOfFailureSync;
    }

    func setNoOfFailureSync(_ noOfFailureSync: Int) {
        self.noOfFailureSync = noOfFailureSync;
    }

    func toString() -> String {
            return "Request{" +
                    "host='" + host + "\'" +
                    ", path='" + path + "\'" +
                    ", type='" + type + "\'" +
                    ", headers=" + headers.toString() +
                    ", body='" + body + "\'" +
                    ", serializer='" + serializer + "\' }"
        }

    func toJSON() -> [String: Any] {
        var request: [String: Any] = [:]
        request["host"] = host
        request["type"] = type
        request["path"] = path
        request["headers"] = headers
        request["serializer"] = serializer
        request["body"] = body
        request["noOfFailureSync"] = noOfFailureSync
        return request
    }

}
